diff --git a/cmd/ipfs/kubo/daemon.go b/cmd/ipfs/kubo/daemon.go
index 0615b26d3..d07fe15f1 100644
--- a/cmd/ipfs/kubo/daemon.go
+++ b/cmd/ipfs/kubo/daemon.go
@@ -64,6 +64,7 @@ const (
 	routingOptionSupernodeKwd  = "supernode"
 	routingOptionDHTClientKwd  = "dhtclient"
 	routingOptionDHTKwd        = "dht"
+	routingOptionBSDHTKwd      = "bsdht"
 	routingOptionDHTServerKwd  = "dhtserver"
 	routingOptionNoneKwd       = "none"
 	routingOptionCustomKwd     = "custom"
@@ -440,6 +441,8 @@ func daemonFunc(req *cmds.Request, re cmds.ResponseEmitter, env cmds.Environment
 		ncfg.Routing = libp2p.DHTOption
 	case routingOptionDHTServerKwd:
 		ncfg.Routing = libp2p.DHTServerOption
+	case routingOptionBSDHTKwd:
+		ncfg.Routing = libp2p.BSDHTOption
 	case routingOptionNoneKwd:
 		ncfg.Routing = libp2p.NilRouterOption
 	case routingOptionCustomKwd:
diff --git a/core/node/libp2p/routingopt.go b/core/node/libp2p/routingopt.go
index 292f49fe4..281dd84a0 100644
--- a/core/node/libp2p/routingopt.go
+++ b/core/node/libp2p/routingopt.go
@@ -14,6 +14,7 @@ import (
 	host "github.com/libp2p/go-libp2p/core/host"
 	"github.com/libp2p/go-libp2p/core/peer"
 	routing "github.com/libp2p/go-libp2p/core/routing"
+	bsdht "github.com/piax/go-byzskip/dht"
 )
 
 type RoutingOptionArgs struct {
@@ -118,6 +119,10 @@ func constructDHTRouting(mode dht.ModeOpt) RoutingOption {
 	}
 }
 
+func BSDHTOption(args RoutingOptionArgs) (routing.Routing, error) {
+	return bsdht.New(args.Host, bsdht.Datastore(args.Datastore), bsdht.RecordValidator(args.Validator), bsdht.BootstrapAddrs(args.BootstrapPeers))
+}
+
 // ConstructDelegatedRouting is used when Routing.Type = "custom"
 func ConstructDelegatedRouting(routers config.Routers, methods config.Methods, peerID string, addrs config.Addresses, privKey string) RoutingOption {
 	return func(args RoutingOptionArgs) (routing.Routing, error) {
